### PR TITLE
feat: [DHIS2-17874] Replace the SnackBar MenuList MenuItem Grow and remaining Icons

### PR DESCRIPTION
**Pull request by henrikmv**
[DHIS2-17874](https://dhis2.atlassian.net/browse/DHIS2-17874)

- Removes MUI components: `SnackBar`, `MenuList`, `MenuItem`, `Grow`, `IconButton`, `Paper`.
- Removes MUI `Icons`

- Uses the dhis2/ui `DropdownButton` to simplify the code.
- Implements style changes to _filter buttons_ to match the chevron in dhis2/ui `DropdownButton`

[DHIS2-17874]: https://dhis2.atlassian.net/browse/DHIS2-17874?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/core_modules/capture-core/components/FeedbackBar/FeedbackBar.component.js [hunk 1]
```diff
@@ -17,110 +14,49 @@ const styles = () => ({
...
+                    </AlertBar>
                 )}
-            </React.Fragment>
-        );
-    }
-}
-Index.displayName = 'FeedbackBar';
+            </AlertStack>
+            {isDialogOpen && (
+                <Modal hide={hide}>
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/FeedbackBar/FeedbackBar.component.js:48]: The modal hiding logic depends on the Redux store feedback and should not be handled internally by the FeedbackBar.component. Can this be [reverted](https://github.com/dhis2/capture-app/pull/3766/files#diff-e5c62664127a93fb774afd3dc3c03ece50718cd03217931b3d88530c015bc485L101) to `hide={!isDialogOpen}`?

File: src/core_modules/capture-core/components/ListView/Filters/FilterButton/ActiveFilterButton.component.js [hunk 2]
```diff
@@ -22,6 +26,7 @@ const getStyles = (theme: Theme) => ({
     },
     clearIcon: {
         color: theme.palette.text.secondary,
+        marginTop: spacers.dp4,
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/ListView/Filters/FilterButton/ActiveFilterButton.component.js:29]: What was the reason for adding this margin? Now it looks to me that the clear icon is a bit lower than the chevron and the filter text 🤔 
- with marginTop

Screenshot 2024-09-05 at 10 39 39

- without marginTop

Screenshot 2024-09-05 at 10 42 25

**Review by simonadomnisoru — COMMENTED** [review]
Hi @henrikmv 
Great idea to use `DropdownButton` and simplify the code 👏 ! Aside from the comments/questions I added, there are only a few places left where Material-UI components are still used:
- MenuItem in [SearchSuggestion.component.js](https://github.com/dhis2/capture-app/blob/hv/feat/DHIS2-17874_ReplaceSnackBarMenuListGrowIcons/src/core_modules/capture-core/components/FormFields/UserField/SearchSuggestion.component.js#L4)
- Paper in [IncompleteSelectionsMessage/index.js](https://github.com/dhis2/capture-app/blob/hv/feat/DHIS2-17874_ReplaceSnackBarMenuListGrowIcons/src/core_modules/capture-core/components/IncompleteSelectionsMessage/index.js#L2)
- IconButton in [ViewEventSection](https://github.com/dhis2/capture-app/blob/hv/feat/DHIS2-17874_ReplaceSnackBarMenuListGrowIcons/src/core_modules/capture-core/components/Pages/ViewEvent/Section/ViewEventSection.component.js#L4) and [Relationships](https://github.com/dhis2/capture-app/blob/hv/feat/DHIS2-17874_ReplaceSnackBarMenuListGrowIcons/src/core_modules/capture-core/components/Relationships/Relationships.component.js#L6)

Can you also migrate these last ones as well? Thank you!

File: src/core_modules/capture-ui/Icons/DropdownChevron.component.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import PropTypes from 'prop-types';
```
[the remark this answers is not in the document]
Comment (henrikmv) [reply src/core_modules/capture-ui/Icons/DropdownChevron.component.js:1]: There is no deep meaning behind it. @Joakim said I should take the **chevron SVG** from the dhis2/ui `DropdownButton` component, and use it in our **filter buttons**. I just copied and pasted the code from the dhis2/ui repo for the _up and down chevron_ without thinking more about it. Of course I will change the code and use flow. Thanks

[the remark this answers is not in the document]
Comment (joakim) [reply src/core_modules/capture-ui/Icons/DropdownChevron.component.js:1]: I must have said that in my sleep 😉

You mentioned the Joakim, but no worries. I agree with my namesake.

[the remark this answers is not in the document]
Comment (henrikmv) [reply src/core_modules/capture-ui/Icons/DropdownChevron.component.js:1]: @joakim Hehe, I've come close to tagging the wrong Joakim a hundred times before. Sorry it actually happened this time.

**Pull request closed without merging**